x/stakeibc/keeper/validator_selection.go

```diff
@@ -71,8 +73,8 @@ func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.Host
 		return validators[i].Weight < validators[j].Weight
 	})
 
-	for i, validator := range hostZone.Validators {
-		if i == len(hostZone.Validators)-1 {
+	for i, validator := range validators {
+		if i == len(validators)-1 {
 			// for the last element, we need to make sure that the allocatedAmt is equal to the finalDelegation
 			targetAmount[validator.GetAddress()] = finalDelegation - allocatedAmt
```

Comment:
Is it worth logging `finalDelegation - allocatedAmt` and the `delegationAmts`? Possibly not if we go look at the `MsgUndelegate`s. On the other hand, it might make comparing the total amount to the validator amounts. Don't feel too strongly either way

Reply:
Yeah, that's essentially logged [here](https://github.com/Stride-Labs/stride/pull/390/files#diff-ba71e9ef6bf1ebcc5243683cbd44405947d8109d7ffb825c7c353d256f7e2179R86)!

`GetTargetValAmtsForHostZone` basically returns a map of the `finalDelegation - allocatedAmt` and `delegationAmts`. And in the link above, we loop through each element in that map